lib/index.js

```diff
@@ -196,13 +197,31 @@ Socket.prototype._stopPollReceive = function () {
  */
 
 Socket.prototype.bind = function (addr) {
-    return this._protect(nn.Bind(this.binding, addr));
+  var eid = nn.Bind( this.binding, addr );
+  if(eid < 0){
+    return this.emit('error', new Error(nn.Err() +': '+this.type+' bind@' + addr));
+  }
+  this.how[addr] = [eid,'bind'];
+  return this.how; //socket's active endpoints
 }
 
 Socket.prototype.connect = function (addr) {
-    return this._protect(nn.Connect(this.binding, addr));
+  var eid = nn.Connect( this.binding, addr )
+  if(eid < 0){
+    return this.emit('error', new Error(nn.Err() +': '+this.type+' connect@' + addr));
+  }
+  this.how[addr] = [eid,'connect'];
+  return this.how; //socket's active endpoints
```

Comment:
instead of saving a tuple of the eid and a string, let's just use two different properties.  Example:

```js
this.connections = {};
this.binds = {};

Socket.prototype.bind = function (addr) {
  var eid = this._protect(nn.Bind(this.binding, addr));
  this.binds[addr] = eid;
  return eid;
};

ditto for connect.  then, we could also probably build from there fixes for #10 and #11, in another PR.
```